### PR TITLE
gpsbabel: update to 1.7.0.

### PR DESCRIPTION
**Pull request by manfredu**

### CODE REVIEW

**Comment by manfredu** [comment]
How can I fix this cross compile error?

**Comment by pullmoll** [comment]
You need to identify what binary `checking loader paths for Qt libs` in `configure` actually tries to run.

Then you can either patch configure to get this information in some way which does not require running a target binary, or you can add `build_helper="qemu"` and try to patch configure to run that `somebinary` with `qemu-${XBPS_TARGET_QEMU_MACHINE}-static somebinary ...`.

**Comment by pullmoll** [comment]
BTW always update your branches with `git pull upstream --rebase --autostash` to avoid the `Merge branch master ..` kind of commits.

If you did not yet set `upstream` for your fork do it like `git remote add upstream https://github.com/void-linux/void-packages.git`

**Comment by ericonr** [comment]
@manfredu ping?

**Comment by manfredu** [comment]
I hope I have some time soon to have a look at it again...

File: srcpkgs/gpsbabel/template [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 license="GPL-2.0-or-later"
 homepage="https://www.gpsbabel.org/"
 distfiles="https://github.com/gpsbabel/${pkgname}/archive/${pkgname}_${version//./_}.tar.gz"
-checksum=8cd740db0b92610abff71e942e8a987df58cd6ca5f25cca86e15f2b00e190704
+checksum=30b186631fb43db576b8177385ed5c31a5a15c02a6bc07bae1e0d7af9058a797
 
 replaces="gpsbabel-gui>=0"
 
+nocross="Configure cannot find Qt library."
+
```
Comment (ericonr) [inline srcpkgs/gpsbabel/template:20]: Adding nocross to a previously cross package is kinda bad :/

**Comment by manfredu** [comment]
> You need to identify the message `checking loader paths for Qt libs` and which binary `configure` actually tries to run there.
> 
> Then you can either patch configure to get this information in some way which does not require running a target binary, or you can add `build_helper="qemu"` and try to patch configure to run that `somebinary` with `qemu-${XBPS_TARGET_QEMU_MACHINE}-static somebinary ...`.

It fails with "loader cannot find Qt library" when cross-compiling in the following part of configure:

```
OCPPFLAGS="$CPPFLAGS"
CPPFLAGS="$CPPFLAGS -I$QT_INC"
OLIBS="$LIBS"
LIBS="$LIBS $QT_LIBS"

OLDFLAGS="$LDFLAGS"
LDFLAGS="$LDFLAGS -Wl,-rpath,$($QMAKE -query QT_INSTALL_LIBS)"


 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
/* end confdefs.h.  */

		#include <QtCore/QtGlobal>
		#include <cstring>
		int main () {
		return strcmp(QT_VERSION_STR, qVersion()) != 0;
		}
_ACEOF
if ac_fn_cxx_try_run "$LINENO"; then :
...
else
  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
as_fn_error $? "loader cannot find Qt library
See \`config.log' for more details" "$LINENO" 5; }
			LDFLAGS="$OLDFLAGS"

fi
```

For the time being I have disabled cross-compiling. Any other suggestion?

I'm still checking why one test is failing with i686 only...

File: srcpkgs/gpsbabel/template (hunk 1 above)
Comment (manfredu) [reply srcpkgs/gpsbabel/template:20]: Sure, but I was not able to make it work. Not sure if it really did work before or the error was just ignored. I assume the later. If someone can help that would be great.

**Comment by manfredu** [comment]
See also https://github.com/GPSBabel/gpsbabel/issues/690

**Comment by ericonr** [comment]
Can you try building with qmake? If that doesn't work, maybe adding #28575 helps

**Comment by manfredu** [comment]
> Can you try building with qmake? If that doesn't work, maybe adding #28575 helps

It's working now, still with configure. Switch to qmake is necessary with the next release. IMHO it is ready to be merged.

File: srcpkgs/gpsbabel/template (hunk 1 above)
Comment (manfredu) [reply srcpkgs/gpsbabel/template]: Fixed

File: srcpkgs/gpsbabel/patches/configure.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+-See \`config.log' for more details" "$LINENO" 5; }
+-else
++#		if test "$cross_compiling" = yes; then :
++#  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
++#$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
++#as_fn_error $? "cannot run test program while cross compiling
++#See \`config.log' for more details" "$LINENO" 5; }
++#else
+@@ -6449 +6447 @@ rm -f core *.core core.conftest.* gmon.o
+-fi
```
Comment (ericonr) [inline srcpkgs/gpsbabel/patches/configure.patch:24]: Ah, nice workaround!

**Pull request merged**